2008-06-03  Jakub Jelinek  <jakub@redhat.com>

	* nscd/nscd_getserv_r.c (__nscd_getservbyport_r): Pass cp
	instead of portstr to nscd_getserv_r.  Patch by
	Roman Kagan <rkagan@mail.ru>.

--- libc/nscd/nscd_getserv_r.c	13 Oct 2007 23:04:28 -0000	1.5
+++ libc/nscd/nscd_getserv_r.c	3 Jun 2008 10:22:52 -0000	1.6
@@ -53,7 +53,7 @@ __nscd_getservbyport_r (int port, const 
   portstr[sizeof (portstr) - 1] = '\0';
   char *cp = _itoa_word (port, portstr + sizeof (portstr) - 1, 10, 0);
 
-  return nscd_getserv_r (portstr, portstr + sizeof (portstr) - cp, proto,
+  return nscd_getserv_r (cp, portstr + sizeof (portstr) - cp, proto,
 			 GETSERVBYPORT, result_buf, buf, buflen, result);
 }
 
